Utils/Game.py

```diff
@@ -1,13 +1,337 @@
+from Utils.Pieces.Pawn import Pawn
+
 class Game:
-    def __init__(self, screen):
-        self.screen = screen
-        self.board = Map(screen)
+    def __init__(self):
+        self.ROW = 8
+        self.COLUMN = 8
+        self.WINDOW_WIDTH = 1100
+        self.WINDOW_HEIGHT = 800
+        self.X_OFFSET = 100
+        self.Y_OFFSET = 100
+        self.SQUARE_SIZE = 75
+        self.BOARD_SIZE = 8 * self.SQUARE_SIZE
+        self.START_BUTTON_X = self.X_OFFSET + self.BOARD_SIZE + 100
+        self.START_BUTTON_Y = self.Y_OFFSET + 120
+        self.UNDO_BUTTON_X = self.X_OFFSET + self.BOARD_SIZE + 100
+        self.UNDO_BUTTON_Y = self.Y_OFFSET + 3 * self.SQUARE_SIZE + 150
+        self.BUTTON_WIDTH = 200
+        self.BUTTON_HEIGHT = 100
+        self.MATERIAL_CHART_WIDTH = 20
+        self.IMAGES = {}
+        self.WHITE = (255, 255, 255)
+        self.GRAY = (128, 128, 128)
+        self.RED = (127, 0, 0, 128)
+        self.DARK_RED = (255, 0, 0, 128)
+        self.GREEN = (0, 128, 0, 128)
+        self.BLACK = (0, 0, 0, 255)
+        self.LIGHT_GRAY = (169, 169, 169)
+        self.BEIGE = (255, 246, 231)
+        self.fps = 30
+        self.screen = pygame.display.set_mode((self.WINDOW_WIDTH, self.WINDOW_HEIGHT), )
+        self.gb = mp.Map(8, 8)
+        self.history = []
+        self.captured_pieces = []
+        self.moves = None
+        self.attack_moves = None
+        self.mouse_down = False
+        self.running = True
         self.selected_piece = None
-        self.selected_piece_moves = []
-        self.selected_piece_attack_moves = []
-        self.selected_original_position = None
-        self.dragged_piece = None
-        self.mouse_x = 0
-        self.mouse_y = 0
-        
\ No newline at end of file
+        self.original_pos = None
+        self.mouse_down = False
+        self.letters = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H']
+        self.fps_counter = 0
+        self.mate = False
+        self.stalemate=False
+        self.promoting_pieces = None
+
+    # reinitializes board after clicking start
+    def reinitialize(self):
+        self.gb = mp.Map(8, 8)
+        self.history = []
+        self.captured_pieces = []
+        self.moves = None
+        self.attack_moves = None
+        self.mouse_down = False
+        self.running = True
+        self.selected_piece = None
+        self.original_pos = None
+        self.mouse_down = False
+        self.letters = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H']
+        self.fps_counter = 0
+        self.mate = False
+        self.stalemate=False
+        self.promoting_pieces = None
+
+    def load_images(self):
+        pieces = ['wR', 'wN', 'wB', 'wQ', 'wK', 'wP', 'bR', 'bN', 'bB', 'bQ', 'bK', 'bP']
+        for piece in pieces:
+            self.IMAGES[piece] = pygame.transform.scale(pygame.image.load('Utils/Images/' + piece + '.png'),
+                                                        (self.SQUARE_SIZE, self.SQUARE_SIZE))
+
+    def draw_board(self):
```

Comment:
All methods should contain docstrings description of parameters and return values.

Example:

```
    def download_blob_to_filepath(
        self, container_name: str, blob_name: str, file_path: str = ""
    ) -> None:
        """
        Downloads a specific blob from the specified container to a local file path.

        :param container_name: The name of the Azure Blob Storage container from which to download.
        :param blob_name: The name of the blob to download.
        :param file_path: The local file path where the blob should be saved.
        """
        container_client = self.blob_service_client.get_container_client(container_name)
        blob_client = container_client.get_blob_client(blob_name)
        destination_path = os.path.join(file_path, blob_name)

        with open(destination_path, "wb") as download_file:
            download_stream = blob_client.download_blob()
            download_file.write(download_stream.readall())
 ```